runtime/src/in_mem_accounts_index.rs

```diff
@@ -465,43 +467,69 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
     }
 
     /// modifies slot_list
-    /// any entry at 'slot' is replaced with 'account_info'.
+    /// any entry at 'slot' or slot 'other_slot' is replaced with 'account_info'.
     /// or, 'account_info' is appended to the slot list if the slot did not exist previously.
     /// returns true if caller should addref
     fn update_slot_list(
```

Comment:
I think the impl looks good here. Again, and as you also pointed out, I dunno about all the cases for doing/not doing a deref when also found at `other_slot`.

Reply:
reworked the loop and improved tests